api-server.py

```diff
@@ -140,8 +185,57 @@ def main():
         endpoint_rest = "tcp:port={0}:interface={1}".format(args.port_rest, host)
         endpoints.serverFromString(reactor, endpoint_rest).listen(Site(api_server_rest.app.resource()))
 
-    app = Klein()
-    app.run(host, 9999)
+    app = ApiKlein()
+    app.run(host, 9999, syslog_facility=syslog_facility)
+
+    # After the reactor is stopped, gracefully shutdown the database.
+    logger.info("Shutting down - closing databases...")
+    NotificationDB.close()
+    Blockchain.Default().Dispose()
+    NodeLeader.Instance().Shutdown()
+
+
+class ApiKlein(Klein):
```

Comment:
The reason I had to create `ApiKlein` is because the standard `Klein.run()` method gives you no way to customize the logging behavior. It forces you to use _either_ `stdout` or a log file. We needed more control over logging, which is why I extended the class. If there is an alternative approach you can think of, great! Let me know...

I don't love doing this kind of thing (overriding and copying 75% of the implementation), but we're beholden to how `Klein` is designed...

Reply:
Understand! Perhaps we should think of taking that into the `neo.api` utilities somewhere